### PR TITLE
Follow-up to OCI download PR merged

### PR DESCRIPTION
**Pull request by booxter**
This addresses comments from https://github.com/instructlab/instructlab/pull/1756 ; plus it optimizes the way symlinks are created, a bit.

### CODE REVIEW

File: src/instructlab/model/download.py [hunk 1]
```diff
@@ -135,11 +135,21 @@ class OCIDownloader(Downloader):
     We are leveraging OCI v1.1 for this functionality
     """
 
-    def __init__(self, repository: str, release: str, download_dest: str, ctx) -> None:
-        super().__init__(
-            repository=repository, release=release, download_dest=download_dest
-        )
-        self.ctx = ctx
+    @staticmethod
+    def _load_json(file_path: Path):
```
Comment (jaideepr97) [inline src/instructlab/model/download.py:139]: @booxter couldn't this generic function to read json files be used elsewhere in the project as well? That was the main reason I'd put it in the `utils.py` file

File: src/instructlab/model/download.py [hunk 2]
```diff
@@ -149,35 +159,35 @@ def _build_oci_model_file_map(self, oci_model_path: str) -> dict:
...
-                    match = _extract_SHA(manifest["digest"])
+                    match = self._extract_SHA(manifest["digest"])
+                    break
 
             if match:
                 index_hash = match.group(1)
             else:
                 raise ValueError(
-                    f"could not find hash for index file at: {oci_model_path}"
+                    f"failed to find image hash in the index file: {oci_model_path}"
```
Comment (jaideepr97) [inline src/instructlab/model/download.py:176]: This is more for my learning, but is it correct to refer to this file as the "image"? It references one of the blobs that represents the manifest that contains the mapping between other blobs and their respective filenames

**Review by jaideepr97 — COMMENTED** [review]
Thanks for the follow up PR @booxter 
Just a couple things from me

File: src/instructlab/model/download.py (hunk 1 above)
Comment (booxter) [reply src/instructlab/model/download.py:139]: That may be true (though it's not used anywhere else right now). Then the name of the function should not be private (remove `_`)?

File: src/instructlab/model/download.py (hunk 2 above)
Comment (booxter) [reply src/instructlab/model/download.py:176]: Oh that may well be that I screwed this one up. The intent here was to say it's not a hash *for* but *in* the index file. The `image` part is wrong - I will drop it.

File: src/instructlab/model/download.py (hunk 1 above)
Comment (jaideepr97) [reply src/instructlab/model/download.py:139]: That would be my preference anyway
I can think of a couple other places we could use this function

Comment (booxter) [reply src/instructlab/model/download.py:139]: I will return the load_json one back; and will leave extract_sha in place since it's very specific to OCI.

**Review by jaideepr97 — APPROVED** [review]
LGTM thanks @booxter

**Review by nathan-weinberg — APPROVED** [review]

**Pull request merged**